runtime/src/message_processor.rs

```diff
@@ -665,7 +665,21 @@ impl MessageProcessor {
         rent_collector: &RentCollector,
         log_collector: Option<Rc<LogCollector>>,
         executors: Rc<RefCell<Executors>>,
+        instruction_index: usize,
     ) -> Result<(), InstructionError> {
+        // Fixup the special instructions key if present
+        // before the account pre-values are taken care of
+        for (i, key) in message.account_keys.iter().enumerate() {
+            if solana_sdk::sysvar::instructions::check_id(key) {
+                let mut mut_account_ref = accounts[i].borrow_mut();
+                solana_sdk::sysvar::instructions::store_current_instruction(
```

Comment:
@sakridge (first and foremost, sorry for late review....)

I think this must be gated as well like this: https://github.com/solana-labs/solana/pull/11943/files#diff-0104dfa96426a3a49428815872a0356eR153

What if we are seeing a meeage with account_keys of `1111111111111111111111111111111`, **before** `solana_sdk::sysvar::instructions::is_enabled`, I think we will end up writing `u16` into empty `account.data` (due to `Account::default()`), which causes bounds-check error at runtime.

Reply:
No problem. Thanks for looking at it. I'll create a follow-up PR.